projects/anomaly-detection-stack/layers/monitoring.yml

```diff
@@ -4,29 +4,42 @@
   connection: local
   roles:
     - role: aws
-      ec2_tags:
-        Name: "smart-monitoring"
-      ec2_region: eu-central-1
-      ec2_image_id: ami-26c43149
       ec2_key_name: "eu-smartcat"
       ec2_security_group: "anomaly-stack"
+      ec2_image_id: ami-26c43149
       ec2_instance_type: m4.xlarge
+      ec2_region: eu-central-1
       ec2_count: 1
+      ec2_volumes:
+        - device_name: /dev/sda1
+          volume_type: gp2
+          volume_size: 30
+          delete_on_termination: true
+        - device_name: /dev/sdh
+          volume_type: gp2
+          volume_size: 100
+          delete_on_termination: true
+      ec2_tags:
+        Name: "smart-monitoring"
       ec2_group: smart-monitoring
 
 - name: Setup instance
   hosts: smart-monitoring
   gather_facts: False
-  tags: ['aws-setup', 'aws-start']
   remote_user: ubuntu
   become: yes
   become_method: sudo
   pre_tasks:
     - name: Gathering facts
       setup:
+      tags: ['aws-setup', 'aws-start']
+    - name: Run filesystem setup
+      include: "{{ playbook_tasks }}/monitoring-os-setup.yml"
+      when: tags is defined
+      tags: ['aws-setup']
```

Comment:
If I am correct this should run only once? Do we want it on aws-start when we are spinning already created instance @mgobec ?